core/services/job/kv_orm.go

```diff
@@ -0,0 +1,69 @@
+package job
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/jmoiron/sqlx/types"
+
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
+)
+
+// KVStore is a simple KV store that can store and retrieve serializable data.
+//
+//go:generate mockery --quiet --name KVStore --output ./mocks/ --case=underscore
+type KVStore interface {
+	Store(key string, val interface{}) error
+	Get(key string, dest interface{}) error
+}
+
+type jobKVStore struct {
+	jobID int32
+	q     pg.Q
+	lggr  logger.SugaredLogger
+}
+
+var _ KVStore = (*jobKVStore)(nil)
+
+func NewJobKVStore(jobID int32, db *sqlx.DB, cfg pg.QConfig, lggr logger.Logger) KVStore {
```

Comment:
Go styling nit: it's usually considered idiomatic to "return structs, accept interfaces" -- in this case I wouldn't be afraid to have the return type be a *jobKVStore and then whereever you use the struct, accept a KVStore instead.

This helps in eg. tests where you may well want to access private methods

Reply:
Curious, my linter moans at me for returning private structs from public functions saying it "can be annoying to use". I often return interfaces for this precise reason.

Do you have a link to more info on this go idiom?

Reply:
Typically you do not declare an interface alongside the struct, and export the struct instead.

Reply:
https://medium.com/@cep21/preemptive-interface-anti-pattern-in-go-54c18ac0668a

Reply:
Also, returning structs allows you to evolves the API in a compatible way: https://go.dev/blog/module-compatibility